x/auth/vesting/client/cli/tx.go

```diff
@@ -79,3 +88,175 @@ timestamp.`,
+	}
+
+	startTime := data.StartTime
+	periods := make([]types.Period, len(data.Periods))
+
+	for i, p := range data.Periods {
+		amount, err := sdk.ParseCoinsNormalized(p.Coins)
+		if err != nil {
+			return 0, nil, err
+		}
+		if p.Length < 1 {
+			return 0, nil, fmt.Errorf("invalid period length of %d in period %d, length must be greater than 0", p.Length, i)
+		}
+
+		periods[i] = types.Period{Length: p.Length, Amount: amount}
+	}
+
+	return startTime, periods, nil
+}
+
+// NewMsgCreateClawbackVestingAccountCmd returns a CLI command handler for creating a
+// MsgCreateClawbackVestingAccount transaction.
+func NewMsgCreateClawbackVestingAccountCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "create-clawback-vesting-account [to_address]",
+		Short: "Create a new vesting account funded with an allocation of tokens, subject to clawback.",
+		Long: `Must provide a lockup periods file (--lockup), a vesting periods file (--vesting), or both.
+If both files are given, they must describe schedules for the same total amount.
+If one file is omitted, it will default to a schedule that immediately unlocks or vests the entire amount.
+The described amount of coins will be transferred from the --from address to the vesting account.
+Unvested coins may be "clawed back" by the funder with the clawback command.
+Coins may not be transferred out of the account if they are locked or unvested, but may be staked.
+Staking rewards are subject to a proportional vesting encumbrance.
+
+A periods file is a JSON object describing a sequence of unlocking or vesting events,
+with a start time and an array of coins strings and durations relative to the start or previous event.`,
+		Example: `Sample period file contents:
+{
+  "start_time": 1625204910,
+  "periods": [
+    {
+      "coins": "10test",
+      "length": 2592000 // 30 days
+    },
+    {
+      "coins": "10test",
+      "length": 2592000 // 30 days
+    }
+  ]
+}
+`,
+		Args: cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx, err := client.GetClientTxContext(cmd)
+			if err != nil {
+				return err
+			}
+
+			toAddr, err := sdk.AccAddressFromBech32(args[0])
+			if err != nil {
+				return err
+			}
+
+			lockupFile, _ := cmd.Flags().GetString(FlagLockup)
+			vestingFile, _ := cmd.Flags().GetString(FlagVesting)
+			if lockupFile == "" && vestingFile == "" {
+				return fmt.Errorf("must specify at least one of %s or %s", FlagLockup, FlagVesting)
+			}
+
+			var (
+				lockupStart, vestingStart     int64
+				lockupPeriods, vestingPeriods []types.Period
+			)
+			if lockupFile != "" {
+				lockupStart, lockupPeriods, err = readScheduleFile(lockupFile)
+				if err != nil {
+					return err
+				}
+			}
+			if vestingFile != "" {
+				vestingStart, vestingPeriods, err = readScheduleFile(vestingFile)
+				if err != nil {
+					return err
+				}
+			}
+
+			commonStart, _ := types.AlignSchedules(lockupStart, vestingStart, lockupPeriods, vestingPeriods)
```

Comment:
I totally agree with you that we can reduce this to be single computation instead of making this computation x2. However, since `startTime` is no longer computed in `tx.go`, changing proto struct so that `MsgCreateClawback` passes lockup start time, vesting start time.

Reply:
What do you mean by it's no longer competed in `tx.go`? Seems that it still is: https://github.com/osmosis-labs/cosmos-sdk/blob/887cbb668b611972f346ae64f43450abe5c09568/x/auth/vesting/client/cli/tx.go#L199

Reply:
i don't think we need to be worried about duplicate computation.

Reply:
It's not for performance reasons. I don't see a reason for passing around fields from cli when they can be computed directly in the vesting account